apps/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -329,41 +328,60 @@ export class RushProjectConfiguration {
-              if (overlapsWithOwnOperation) {
-                if (overlappingOperationNames.length === 1) {
-                  terminal.writeErrorLine(
-                    `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
-                      `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
-                      `operation ${operationName} overlaps with another folder name in the same operation.`
-                  );
-                } else {
-                  const otherOperationNames: string[] = overlappingOperationNames.filter(
-                    (overlappingOperationName) => overlappingOperationName !== operationName
-                  );
-                  terminal.writeErrorLine(
-                    `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
-                      `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
-                      `operation ${operationName} overlaps with other folder names in the same operation and with ` +
-                      `folder names in the following other operations: ${otherOperationNames.join(', ')}.`
-                  );
+      // For each phased command, check if any of its phases' output folders overlap.
+      for (const [, command] of repoCommandLineConfiguration.commands) {
+        if (command.commandKind === 'phased') {
+          const phasesOverlappingPathAnalyzer: OverlappingPathAnalyzer<string> =
+            new OverlappingPathAnalyzer<string>();
+
+          for (const operationName of command.phases) {
+            const operationSettings: IOperationSettings | undefined =
+              operationSettingsByOperationName.get(operationName);
+            if (operationSettings) {
+              if (operationSettings.outputFolderNames) {
+                for (const outputFolderName of operationSettings.outputFolderNames) {
+                  const overlappingOperationNames: string[] | undefined =
+                    phasesOverlappingPathAnalyzer.addPathAndGetFirstEncounteredLabels(
+                      outputFolderName,
+                      operationName
+                    );
+                  if (overlappingOperationNames) {
+                    const overlapsWithOwnOperation: boolean =
+                      overlappingOperationNames?.includes(operationName);
+                    if (overlapsWithOwnOperation) {
+                      if (overlappingOperationNames.length === 1) {
+                        terminal.writeErrorLine(
+                          `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
+                            `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
+                            `operation "${operationName}" overlaps with another folder name in the same operation.`
```

Comment:
> Error: The "rush-project.json" config file defines an operation "_phase:build" with overlapping
> "outputFolderNames" paths: "lib", "lib/public", "assets", "assets/public".

Reply:
If this error is printed, then it might be okay to ignore the cross-command overlap.  After the person fixes the simple problem, it's not a big deal if they see a new error about the other problem, rather than trying to mix two separate problems into one super complicated sentence.

Reply:
Updated to

```
The "config/rush-project.json" config file defines an operation "_phase:build" with a path ("lib") in
"outputFolderNames" that overlaps with another path in the same operation.
```

Reply:
The phrase "with another path in the same operation" seems slightly weird to me.  The two logical cases make sense to somebody reading the source code, but for an end user, it sounds a bit like "Two hands are clapping that belong to the same person." 😋

Here's another possible iteration:
```
The project "myproject" has a "config/rush-project.json" configuration that defines an operation
with overlapping paths in the "outputFolderNames" list. The operation is "_phase:build", and the
conflicting path is "lib".
```